### PR TITLE
Convert Reservation net price fields into properties

### PR DESCRIPTION
**Pull request by ranta**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Convert Reservation net price fields into properties
- This should not require changes from the frontend, as the fields should be simply ignored if they are sent

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests
- Might be good to manually test as well just in case :D

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3485](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3485)

[TILA-3485]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3485?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by matti-lamppu — APPROVED** [review]

File: reservations/models/reservation.py [hunk 1]
```diff
@@ -212,6 +210,14 @@ class Meta:
     def __str__(self) -> str:
         return f"{self.name} ({self.type})"
 
+    @property
+    def price_net(self) -> Decimal:
+        return round_decimal(self.price / (1 + self.tax_percentage_value / Decimal(100)), 6)
```
Comment (vergama) [inline reservations/models/reservation.py:215]: I think we can round to the usual 2 decimal places when this is now an ephemeral and no longer a stored value.

File: reservations/models/reservation.py [hunk 2]
```diff
@@ -212,6 +210,14 @@ class Meta:
...
+
+    @property
+    def non_subsidised_price_net(self) -> Decimal:
+        return round_decimal(self.non_subsidised_price / (1 + self.tax_percentage_value / Decimal(100)), 6)
```
Comment (vergama) [inline reservations/models/reservation.py:219]: Same as above with the (subsidised) net price, 2 decimal prices would be appropriate.

File: merchants/verkkokauppa/helpers.py [hunk 3]
```diff
@@ -69,8 +69,8 @@ def get_meta_label(key: str, reservation: Reservation) -> str:
 def get_verkkokauppa_order_params(reservation: Reservation) -> CreateOrderParams:
     reservation_unit = reservation.reservation_unit.first()
     quantity = 1  # Currently, we don't support quantities larger than 1
-    price_net = round_decimal(Decimal(quantity * reservation.price_net), 2)
-    price_vat = round_decimal(Decimal(quantity * reservation.price_net * (reservation.tax_percentage_value / 100)), 2)
+    price_net = round_decimal(quantity * reservation.price_net, 2)
```
Comment (vergama) [inline merchants/verkkokauppa/helpers.py:72]: The formula should be:
1. gross price = quantity * reservation.price # e.g. 1 * 123.45 €
2. net price = round_decimal(gross price / (1 + reservation.tax_percentage_value / 100), 2) # e.g. 123.45 / 1.255 = 98.366... ~= 98.37 € -- or in practice reservation.price_net as long as quantity is always 1
3. vat amount = gross - net price # e.g. 123.45 - 98.37 = 25.08 €

**Review by vergama — CHANGES_REQUESTED** [review]
Two comments: these go back to the old code but let's fix them now.

**Review by vergama — APPROVED** [review]

**Pull request merged**